clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -560,21 +513,25 @@ void ConstraintBuilderConsumer::HandleTranslationUnit(ASTContext &C) {
     else
       errs() << "Analyzing\n";
   }
-  TypeVariableBindingsMapT TypeVariableBindings;
-  GlobalVisitor GV = GlobalVisitor(&C, Info, TypeVariableBindings);
+
+
+  VariableAdderVisitor VAV = VariableAdderVisitor(&C, Info);
```

Comment:
Per our discussion on Zoom, we should think about how to describe this visitor, and ensure it's complete. E.g., is it the only think that will be adding to the `Variables` map? If so, we can ensure that follow-on visitors never do, with appropriate `assert`s.

Reply:
It's currently the only place that `addVariable` is called, but the `Variables` map is also modified through `getPersistentConstraintVars` which is called in a few other places in `ConstraintResolver`.

Reply:
We could set some a in `ProgramInfo` to cause an assert to fail if `addVariable` is called after the variable adding visitor. We should also look at how calls to `getPersistentConstraintVars` (and anything else that touch the variables map)  might be moved into this visitor.

Reply:
Yes, sounds good.